kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.axonframework.extensions.kotlin.serialization
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.PolymorphicSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.builtins.nullable
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.polymorphic
+import kotlinx.serialization.modules.subclass
+import org.axonframework.eventhandling.GapAwareTrackingToken
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken
+import org.axonframework.eventhandling.MergedTrackingToken
+import org.axonframework.eventhandling.MultiSourceTrackingToken
+import org.axonframework.eventhandling.ReplayToken
+import org.axonframework.eventhandling.TrackingToken
+import org.axonframework.eventhandling.scheduling.ScheduleToken
+import org.axonframework.eventhandling.scheduling.java.SimpleScheduleToken
+import org.axonframework.eventhandling.scheduling.quartz.QuartzScheduleToken
+import org.axonframework.eventhandling.tokenstore.ConfigToken
+import org.axonframework.messaging.responsetypes.InstanceResponseType
+import org.axonframework.messaging.responsetypes.MultipleInstancesResponseType
+import org.axonframework.messaging.responsetypes.OptionalResponseType
+import org.axonframework.messaging.responsetypes.ResponseType
+import kotlin.reflect.KClass
+
+private val trackingTokenSerializer = PolymorphicSerializer(TrackingToken::class).nullable
```

Comment:
Reading the documentation of the `PolymorphicSerializer` constructor, I read the following:

```txt
...
 * To avoid the most common security pitfalls and reflective lookup (and potential load) of an arbitrary class,
 * all serializable implementations of any polymorphic type must be [registered][SerializersModuleBuilder.polymorphic]
 * in advance in the scope of base polymorphic type, efficiently preventing unbounded polymorphic serialization
 * of an arbitrary type.
...
```

So, I assume that when this `trackingTokenSerializer` is instantiated all implementation have been registered, and novel implementation can be included by users as well?